src/network/clientpackethandler.cpp

```diff
@@ -1240,28 +1241,69 @@ void Client::handleCommand_HudSetParam(NetworkPacket* pkt)
 
 void Client::handleCommand_HudSetSky(NetworkPacket* pkt)
 {
-	std::string datastring(pkt->getString(0), pkt->getSize());
-	std::istringstream is(datastring, std::ios_base::binary);
+	SkyboxParams skybox;
+	u16 texture_count;
+	std::string texture;
+
+	*pkt >> skybox.bgcolor >> skybox.type >> skybox.clouds >>
```

Comment:
Beginning from `skybox.clouds` it's not backwards compatible. The absolute minimal compatibility code would look as follows:
```C++
if (m_proto_ver < 39) {
	// Skybox not applied on older servers
	return;
}
// New skybox format
*pkt >> skybox.bgcolor >> skybox.type >> ....
```

Reply:
Would leaving a warning in console be wise enough or actually going the whole way and keeping compatibility?

Reply:
Full compatibility is definitely needed here, you won't want the skybox to go missing just because you connected to an older server.

Reply:
Coming back to this, should an old packet keep the classic intended set_sky behaviour, as in hiding the sun, moon and stars depending on skybox mode?

Reply:
It should come reasonably close to the behavior the server actually intended.
That's a yes.

Reply:
It's fully functional with 5.1.0-dev from local testing, works on Inside The Box.